src/server/dcp_server/utils.py

```diff
@@ -212,21 +239,71 @@ def create_patch_dataset(imgs, masks_classes, masks_instances, noise_intensity,
     the max cell size to define the patch size. All patches and masks should then be returned
     in the same format as imgs and masks (same type, i.e. check if tensor or np.array and same 
     convention of dims, e.g.  CxHxW)
+    include_mask(bool) : Flag indicating whether to include the mask along with patches. 
     '''
 
     if max_patch_size is None:
         max_patch_size = np.max([find_max_patch_size(mask) for mask in masks_instances])
         
 
-    patches, labels = [], []
+    patches, patch_masks, labels = [], [], []
     for img, mask_class, mask_instance in zip(imgs,  masks_classes, masks_instances):
         # mask_instance has dimension WxH
         # mask_class has dimension WxH
-        patch, _, label = get_centered_patches(img,
+        patch, patch_mask, _, label = get_centered_patches(img,
                                             mask_instance,
                                             max_patch_size, 
                                             noise_intensity=noise_intensity,
-                                            mask_class=mask_class)
+                                            mask_class=mask_class,
+                                            )
         patches.extend(patch)
+        patch_masks.extend(patch_mask)
         labels.extend(label) 
-    return patches, labels
\ No newline at end of file
+    return patches, patch_masks, labels
+
+
+def get_shape_features(img, msk):
+
+    msk = 255 * ((msk) > 0).astype(np.uint8)
+
+    image = sitk.GetImageFromArray(img.squeeze())
```

Comment:
is it necessary to use sitk function or could is same functionality covered by an existing library?

Reply:
Unfortunately, according to PyRadiomics, it can only be done using SimpleITK (sitk) or by saving the image beforehand and then reading it, but the latter is less preferable